synapse/rest/synapse/client/pick_username.py

```diff
@@ -113,11 +113,19 @@ async def _async_render_POST(self, request: SynapseRequest):
 
         try:
             localpart = parse_string(request, "username", required=True)
+            use_display_name = parse_boolean(request, "use_display_name", default=False)
+
+            try:
+                emails_to_use = [
+                    val.decode("utf-8") for val in request.args.get(b"use_email", [])
+                ]  # type: List[str]
```

Comment:
Does this list need to be checked against the list that was given to the user? (Right now it seems like it could be used to inject *any* email?)

Reply:
that is (hopefully) checked in the handler.

Reply:
specifically, here: https://github.com/matrix-org/synapse/pull/9277/commits/f96c0dec56670885428823b7c6ee225ca10cd609#diff-05be763854c3b4a72fd8c85ffb5a6b3a10da786f46c53ca4b64d1f150d8fc992R744

Reply:
Ah, OK! Just done in a different layer than expected!

Reply:
Arg, #6739 adds a `parse_list_from_args` which would be useful here.

Reply:
that feels a bit overengineered, if I'm honest.